columnflow/tasks/framework/base.py

```diff
@@ -508,59 +518,59 @@ def modify_param_values(cls, params):
         shifts, upstream_shifts = cls.get_known_shifts(config_inst, params)
 
         # actual shift resolution: compare the requested shift to known ones
-        if not requested_effective_shift or requested_effective_shift == law.NO_STR:
+        # task_shift -> the requested shift if implemented by the task itself, else nominal
+        # shift      -> the requested shift if implemented by this task
+        #               or an upsteam task (== effective shift), else nominal
+        if requested_task_shift in (None, law.NO_STR):
             if requested_shift in shifts:
                 params["shift"] = requested_shift
-                params["effective_shift"] = requested_shift
+                params["task_shift"] = requested_shift
             elif requested_shift in upstream_shifts:
-                params["effective_shift"] = "nominal"
+                params["shift"] = requested_shift
+                params["task_shift"] = "nominal"
             else:
                 params["shift"] = "nominal"
-                params["effective_shift"] = "nominal"
+                params["task_shift"] = "nominal"
 
-        # store references
-        params["requested_shift_inst"] = config_inst.get_shift(params["shift"])
-        params["effective_shift_inst"] = config_inst.get_shift(params["effective_shift"])
+        # store references (not the meaning of effective and task shifts here)
+        params["effective_shift_inst"] = config_inst.get_shift(params["shift"])
+        params["shift_inst"] = config_inst.get_shift(params["task_shift"])
```

Comment:
One minor thing: the nomenclature here is a bit confusing, since `shift_inst` does not correspond to the `shift`parameter. Maybe we could rename the `shift_inst`to `task_shift_inst`?

Reply:
Thanks for the review!

This exact question is also still on my mind. I actually called it `task_shift_inst` first (and had a `@property` named `shift_inst` as an alias), but then `self.task_shift_inst` or even `self.task.task_shift_inst` in (e.g.) producers or the ml model seemed a bit redundant. What do you think?

Reply:
Also @dsavoiu  :)